lib/adapter.js

```diff
@@ -26,7 +26,8 @@ var metrics = {
 
 var internal = {
   on: internal_on, // XXX(sam) where is this used?
-  supports: {watchdog: false} // XXX(sam) probably not needed
+  supports: {watchdog: process.platform === 'linux' &&
+    (process.arch === 'ia32' || process.arch === 'x64') }
```

Comment:
I don't think this check is sufficient, see https://github.com/RuntimeTools/appmetrics/pull/290/files#r78633579

I'll read https://github.com/RuntimeTools/appmetrics/pull/290 more closely, it should be possible to determine from the appmetrics addon whether watchdog is or is not supported, so that all platform dependencies are in the c++, and js just checks to see if the feature exists.